internal/auth/option.go

```diff
@@ -87,3 +89,9 @@ func WithAnonymousUserNotAllowed(notAllowed bool) Option {
 		o.withAnonymousUserNotAllowed = notAllowed
 	}
 }
+
+func WithResource(resource *perms.Resource) Option {
```

Comment:
Mind adding docs to this exported func?   Also, it appears there aren't unit tests for these options.  Can you start a new pattern and add one for this new option, which will align with what the other repos are doing?

Reply:
Done.